docs/proposals/node-allocatable.md

```diff
@@ -0,0 +1,177 @@
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/kubemark.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Node Allocatable Resources
+
+**Issue:** https://github.com/kubernetes/kubernetes/issues/13984
+
+## Overview
+
+Currently Node.Status has Capacity, but no concept of Machine Allocatable. We need additional
+parameters to serve several purposes:
+
+1. [Kubernetes metrics](compute-resource-metrics-api.md) provides "/docker-daemon", "/kubelet",
+   "/kube-proxy", "/system" etc. raw containers for monitoring system component resource usage
+   patterns and detecting regressions. Eventually we want to cap system component usage to a certain
+   limit / request. However this is not currently feasible because 1) docker still uses tons of
+   computing resources and 2) we have not yet defined the minimal system requirements, so we cannot
+   control Kubernetes nodes or know about arbitrary daemons, which can make the system resources
+   unmanageable. Even with a resource cap we cannot do a full resource management on the node, but
+   with the proposed parameters we can mitigate really bad resource overcommits.
+2. For external schedulers (such as mesos, hadoop, etc.) integration, they might want to partition
+   compute resources on a given node, limiting how much Kubelet can use. We should provide a
+   mechanism by which they can query kubelet, and reserve some resources for their own purpose.
+
+### Scope of proposal
+
+This proposal deals with resource reporting through the [`Allocatable` field](#allocatable) for more
+reliable scheduling, and minimizing resource over commitment. This proposal *does not* cover
+resource usage enforcement (e.g. limiting system component usage), and pod eviction (e.g. when
+reservation grows).
+
+## Design
+
+### Definitions
+
+![image](node-allocatable.png)
+
+1. **Machine Capacity** - Already provided as
+   [`NodeStatus.Capacity`](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/HEAD/docs/api-reference/v1/definitions.html#_v1_nodestatus),
+   this is total capacity read from the machine instance, and assumed to be constant.
+2. **Reserved** (proposed) - Compute resources reserved for non-Kubernetes and non-system
```

Comment:
Yesterday when we discussed to remove this since we proposed another exclusive config knob --allocatable to support meso's case, we talked about removing this. Actually I think we still should keep this for those non-kubernetes daemons / processes which are currently running in /system cgroup.